transforms/polus-recycle-vector-plugin/src/main.py

```diff
@@ -1,119 +1,142 @@
-                        datefmt='%d-%b-%y %H:%M:%S')
-    logger = logging.getLogger("main")
-    logger.setLevel(logging.INFO)
-    
-    # Setup the Argument parsing
-    logger.info("Parsing arguments...")
-    parser = argparse.ArgumentParser(prog='main', description='Extract individual fields of view from a czi file.')
+def main(stitch_dir: Path,
+         collection_dir: Path,
+         output_dir: Path,
+         pattern: str):
 
-    parser.add_argument('--stitchDir', dest='stitch_dir', type=str,
-                        help='Stitching vector to recycle', required=True)
-    parser.add_argument('--collectionDir', dest='collection_dir', type=str,
-                        help='Image collection to place in new stitching vector', required=True)
-    parser.add_argument('--stitchRegex', dest='stitch_regex', type=str,
-                        help='Stitching vector regular expression', required=True)
-    parser.add_argument('--collectionRegex', dest='collection_regex', type=str,
-                        help='Image collection regular expression', required=True)
-    parser.add_argument('--groupBy', dest='group_by', type=str,
-                        help='Variables to group within a single stitching vector', required=False)
-    parser.add_argument('--outDir', dest='output_dir', type=str,
-                        help='The directory in which to save stitching vectors.', required=True)
+    if pattern in [None,'.+','.*']:
+        pattern = filepattern.infer_pattern([f.name for f in collection_dir.iterdir()])
+        logger.info(f'Inferred filepattern: {pattern}')
 
-    # Get the arguments
-    args = parser.parse_args()
-    stitch_dir = args.stitch_dir
-    collection_dir = args.collection_dir
-    stitch_regex = args.stitch_regex
-    collection_regex = args.collection_regex
-    group_by = args.group_by
-    output_dir = args.output_dir
-    logger.info('stitch_dir = {}'.format(stitch_dir))
-    logger.info('collection_dir = {}'.format(collection_dir))
-    logger.info('stitch_regex = {}'.format(stitch_regex))
-    logger.info('collection_regex = {}'.format(collection_regex))
-    logger.info('output_dir = {}'.format(output_dir))
-    
     # Parse files in the image collection
-    fp = filepattern.FilePattern(collection_dir,collection_regex)
-    
-    # Process group_by variable
-    if group_by==None:
-        group_by = ''
-    for v in 'xyp':
-        if v not in group_by:
-            group_by += v
-    group_by = group_by.lower()
-    group_by = ''.join([g for g in group_by if g in fp.variables])
-    logger.info('group_by = {}'.format(group_by))
+    fp = filepattern.FilePattern(collection_dir,pattern)
     
-    # Loop through the stitching vectors
+    # Get valid stitching vectors
     vectors = [v for v in Path(stitch_dir).iterdir() if Path(v).name.startswith('img-global-positions')]
+    
+    # Get filepatterns for each stitching vector
```

Comment:
I find this code to be pretty opaque. You are using -1, 1, and None to represent states of `singulars` which you use later. It would be immensely helpful if you add a few comments explaining what's going on. Sample applies to around line 55.